### PR TITLE
[WIP] Reduce test target complexity [and running time]

### PR DESCRIPTION
**Pull request by piotrpMSFT**
I was inspired by @blackdwarf's work to reduce the many targets of 'test restore' into a single sln and single restore invocation. Greatly reduced concept count and a slight perf improvement.

This change eliminates a bunch of duplicate test orchestration code, resulting in modest perf gains and lots of simplification. Included:
- Tests are no longer built before execution. This saves ~7s per test project that is required for `dotnet test` to prove that the test library is up-to-date
- CLI build infra no longer attempts to perform incrementality checks on behalf of `dotnet build`. This was necessary when we were wrapping project.json but is no longer needed.
- Test Assets are also no longer pre-built, saving many cycles
- No more incrementality checks from CLI build system for Test Assets
- Test targets are significantly reduced in complexity [though still complex]

The PR is a WiP. I expect issues due to `/` vs `\`. I'm also not able to test locally due to pending reboot, so will see what CI has to say.

/cc @livarcocc

### CODE REVIEW

File: src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs [hunk 1]
```diff
@@ -346,7 +346,7 @@ public IEnumerable<FileInfo> LoadInventory(FileInfo file)
                 file.Refresh();
                 if (!file.Exists)
                 {
-                    throw new ArgumentException("Inventory file should exist.");
```
Comment (livarcocc) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: Why do we get flipping on this? We had empty, then ArgumentException and back to Empty.

Comment (piotrpMSFT) [inline src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs]: ArgumentException didn't account for rebuild scenarios. This infrastructure is likely getting complex enough to warrant its own tests.

**Comment by piotrpMSFT** [comment]
@MattGertz FYI, this infrastructure PR eliminates a bunch of the custom incremental build logic that CLI depended on while migrating CLI's build from PJ to CSProj. This greatly reduces CLI build complexity and, more importantly, build times. The PR also removes a deprecated and long-running test library [performance validation for project.json system which, after recent re-evaluation, adds little value for MSBuild/SDK scenarios. These will need to be replaced with propper performance tests for csproj in the near future].

The combined benefit was observed in this PR as a reduction from ~33 minute builds to ~18 minute builds. Wohoo!

**Comment by MattGertz** [comment]
:-)

**Comment by DustinCampbell** [comment]
A lot of red in this PR, which is goodness in my book!

**Pull request merged**